src/main/java/com/redhat/cloud/policies/engine/db/repositories/PoliciesRepository.java

```diff
@@ -41,30 +43,68 @@ public List<Policy> getEnabledPolicies(String accountId) {
+        EnabledPolicies enabledPolicies = enabledPoliciesCacheOrgId.computeIfAbsent(orgId, unused -> new EnabledPolicies());
+        LocalDateTime dbLatestUpdate = findLatestUpdateOrgId(orgId);
+        if (enabledPolicies.latestUpdate == null || enabledPolicies.latestUpdate.isBefore(dbLatestUpdate)) {
+            LOGGER.debug("Reloading enabled policies from DB");
+            enabledPolicies.policies = findEnabledPoliciesByOrgId(orgId);
+            enabledPolicies.latestUpdate = dbLatestUpdate;
+        }
+        return enabledPolicies.policies;
+    }
+
     @CacheResult(cacheName = ACCOUNT_LATEST_UPDATE_CACHE_NAME)
     LocalDateTime findLatestUpdate(String accountId) {
         LOGGER.debugf("Finding latest policies update time for account %s", accountId);
-        String hql = "SELECT latest FROM AccountLatestUpdate WHERE accountId = :accountId";
         try {
+            String hql = "SELECT latest FROM AccountLatestUpdate WHERE accountId = :accountId";
             return statelessSessionFactory.getCurrentSession().createQuery(hql, LocalDateTime.class)
                     .setParameter("accountId", accountId)
                     .getSingleResult();
         } catch (NoResultException e) {
-            LOGGER.debugf("No latest policies update time found, using default value", accountId);
+            LOGGER.debug("No latest policies update time found, using default value");
+            return LocalDateTime.MIN;
+        }
+    }
+
+    @CacheResult(cacheName = ORG_ID_LATEST_UPDATE_CACHE_NAME)
+    LocalDateTime findLatestUpdateOrgId(String orgId) {
+        LOGGER.debugf("Finding latest policies update time for orgId %s", orgId);
+        try {
+            String hql = "SELECT latest FROM OrgIdLatestUpdate WHERE orgId = :orgId";
+            return statelessSessionFactory.getCurrentSession().createQuery(hql, LocalDateTime.class)
+                    .setParameter("orgId", orgId)
+                    .getSingleResult();
+        } catch (NoResultException e) {
+            LOGGER.debug("No latest policies update time found, using default value");
             return LocalDateTime.MIN;
         }
     }
 
     private List<Policy> findEnabledPoliciesByAccount(String accountId) {
         String hql = "FROM Policy WHERE accountId = :accountId AND enabled IS TRUE";
         return statelessSessionFactory.getCurrentSession().createQuery(hql, Policy.class)
-            .setParameter("accountId", accountId)
-            .getResultList();
+                .setParameter("accountId", accountId)
+                .getResultList();
+    }
+
+    private List<Policy> findEnabledPoliciesByOrgId(String orgId) {
+        String hql = "FROM Policy WHERE orgId = :orgId AND enabled IS TRUE";
+        return statelessSessionFactory.getCurrentSession().createQuery(hql, Policy.class)
+                .setParameter("orgId", orgId)
+                .getResultList();
     }
 
     @CacheInvalidateAll(cacheName = ACCOUNT_LATEST_UPDATE_CACHE_NAME)
```

Comment:
@gwenneg do we (you) still need that cache?

Reply:
argh again. I meant the clear cache method which seems to be unused.

Reply:
Hmm, it was used from a test initially. Maybe that test was changed and this method is no longer needed.